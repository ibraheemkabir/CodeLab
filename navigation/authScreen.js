import React from 'react';
import {
	AsyncStorage,
	View,
} from 'react-native';
import { graphql } from 'react-apollo'
import gql from 'graphql-tag';


export const FEED_QUERY = gql`
	 query {
    viewer {
      avatarUrl
    }
  }
`;

class AuthLoadingScreen extends React.Component {
	constructor(props) {
		super(props);
		this.auth();
	}
	static navigationOptions = ({ navigation }) => {
		return {
			headerLeft: null,
			headerLeft: <View></View>,
			headerBackTitle: null
		}
	};
	auth = async () => {
		const viewer = await this.props.feedQuery.viewer || null
		const userToken = await AsyncStorage.getItem('GithubStorageKey');
		this.props.navigation.navigate(userToken ? 'List' : 'Main', { itemId: viewer});
	};

	// Render any loading content that you like here
	render() {
		return (
			<View>
			</View>
		);
	}
}

export default graphql(FEED_QUERY, {
	name: 'feedQuery',
	options: {
		fetchPolicy: 'network-only',
	},
})(AuthLoadingScreen)