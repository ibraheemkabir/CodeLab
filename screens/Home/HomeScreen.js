import React, { Component } from 'react';
import { View, StyleSheet, Text, Image } from 'react-native';
import getGithubTokenAsync from '../../utils/githubAuth';
import { Container, Title, Content, Item, Input, Button } from 'native-base';
import { AsyncStorage } from 'react-native';
import { graphql } from 'react-apollo'
import gql from 'graphql-tag';

export const FEED_QUERY = gql`
	 query {
    viewer {
      avatarUrl
    }
  }
`;

class LogoTitle extends React.Component {

	render() {
		console.log(this.props)
		return (
			<View style={{ flex: 1, flexDirection: 'row', paddingLeft: 10}}>
				<Title style={{ color: 'white'}}>CodeLab</Title>
			</View>
		);
	}
}
class HomeScreen extends Component {
	static navigationOptions = {
		headerLeft: <LogoTitle />,
		headerStyle: {
			backgroundColor: '#A02C2D'
		},
		headerBackTitle: null,
	};

	state = {
		loading: false
	}

	async signInAsync() {
		this.setState({loading:true})
		try {
					const token = await getGithubTokenAsync();
					await AsyncStorage.setItem('GithubStorageKey', token);
					const viewer = this.props.feedQuery.viewer || null
					if (token !== null) {
						this.props.navigation.navigate('List', {
							itemId: viewer,
					})
					this.setState({loading:false})
					return token;
				}
		} catch ({ message }) {
			return message
			console.log('hello')
		}
	};

	render() {
		console.log(this.props)
		return (
			<Container>
				{
					this.state.loading && <View style={{ alignItems: 'center',paddingTop: 10 }}><Text><Image source={require('../../assets/images/loading.gif')} /></Text></View>
				}
				<Content style={styles.content}>
					<View style={styles.Image}>
						<Image
							source={require('../../assets/images/robot-dev.png')}
							style={{marginBottom: 15}}
						/>
					</View>					
					<Button 
					onPress={()=>this.signInAsync()}
					id= 'btn'
					style={{ flex: 1, justifyContent: 'center', alignContent: 'center', backgroundColor: '#E18D96'}}
					>
						<Text style={{
							flex: 1, textAlign: 'center', fontWeight: 'bold', color: 'white' }} id='btntext'>
							Sign In with Github
						</Text>
					</Button>
				</Content>
			</Container>
		);
	}
}

const styles = StyleSheet.create({
	content: {
		flex: 1,
		justifyContent: 'center',
		alignContent: 'center',
		padding: 20,
		paddingTop: 0,
		marginBottom: 200
	},
	Input: {
		marginBottom: 20,
		textAlign: 'left'
	},
	Image: {
		flex: 1,
		alignItems: 'center',
		padding: 5
	},
	button: {
		flex: 1, 
		justifyContent: 'center', 
		alignContent: 'center', 
		backgroundColor: '#E18D96'
	}
})

export default graphql(FEED_QUERY, {
	name: 'feedQuery',
	options: {
		fetchPolicy: 'network-only',
	},
})(HomeScreen)
