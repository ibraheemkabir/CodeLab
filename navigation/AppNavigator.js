import { createAppContainer, createStackNavigator } from 'react-navigation';
import HomeScreen from '../screens/Home/HomeScreen';
import ListScreen from '../screens/List/ListScreen';
import ProfileScreen from '../screens/Profile/ProfileScreen';



export default createAppContainer(
  createStackNavigator({
   Main: HomeScreen,
   List: ListScreen,
   Profile: ProfileScreen
  })
);
