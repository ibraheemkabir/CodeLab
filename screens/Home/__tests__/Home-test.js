import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import Home from '../HomeScreen';
import { MockedProvider } from 'react-apollo/test-utils';
import { FEED_QUERY } from '../../../navigation/authScreen';
jest.mock('expo', () => ({
  AuthSession: {
    getRedirectUrl: jest.fn(),
  },
}));
it(`renders correctly`, () => {
  let props = {
    navigation: {
      navigate: jest.fn(),
      setParams: jest.fn()
    },
    AuthSession: jest.fn()
  }

  const home = renderer.create(
    <MockedProvider addTypename={false}>
      <Home {...props} />
    </MockedProvider>
  );

  expect(home).toMatchSnapshot();
});


