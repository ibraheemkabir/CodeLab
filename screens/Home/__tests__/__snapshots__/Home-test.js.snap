// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "height": 1334,
    }
  }
>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentContainerStyle={
      Array [
        Object {
          "padding": undefined,
        },
        undefined,
      ]
    }
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutomaticScroll={true}
    enableOnAndroid={false}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardShouldPersistTaps="handled"
    keyboardSpace={0}
    onScroll={[Function]}
    resetKeyboardSpace={[Function]}
    resetScrollToCoords={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollIntoView={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
        },
        Object {
          "alignContent": "center",
          "flex": 1,
          "justifyContent": "center",
          "marginBottom": 200,
          "padding": 20,
          "paddingTop": 0,
        },
      ]
    }
    update={[Function]}
    viewIsInsideTabBar={false}
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "padding": 5,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "marginBottom": 15,
            }
          }
        />
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "alignSelf": "flex-start",
            "backgroundColor": "#E18D96",
            "borderBottomWidth": null,
            "borderColor": "#007aff",
            "borderLeftWidth": null,
            "borderRadius": 5,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "borderWidth": undefined,
            "elevation": 2,
            "flex": 1,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "center",
            "opacity": 1,
            "paddingBottom": 6,
            "paddingTop": 6,
            "shadowColor": undefined,
            "shadowOffset": undefined,
            "shadowOpacity": undefined,
            "shadowRadius": undefined,
          }
        }
      >
        <Text
          id="btntext"
          style={
            Object {
              "color": "white",
              "flex": 1,
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          Sign In with Github
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
